modules/core/src/main/java/org/apache/ignite/internal/processors/performancestatistics/FilePerformanceStatisticsWriter.java

```diff
@@ -0,0 +1,643 @@
+
+        if (seg == null)
+            return;
+
+        ByteBuffer buf = seg.buffer();
+
+        writeIgniteUuid(buf, sesId);
+        buf.putLong(queuedTime);
+        buf.putLong(startTime);
+        buf.putLong(duration);
+        buf.put(timedOut ? (byte)1 : 0);
+
+        seg.release();
+    }
+
+    /**
+     * Reserves buffer's write segment.
+     *
+     * @return Buffer's write segment or {@code null} if not enought space or writer stopping.
+     */
+    private SegmentedRingByteBuffer.WriteSegment reserveBuffer(OperationType type, int size) {
+        FileWriter fileWriter = this.fileWriter;
+
+        // Writer stopping.
+        if (fileWriter == null)
+            return null;
+
+        SegmentedRingByteBuffer.WriteSegment seg = fileWriter.writeSegment(size + /*type*/ 1);
+
+        if (seg == null) {
+            fileWriter.logSmallBufferMessage();
+
+            return null;
+        }
+
+        // Ring buffer closed (writer stopping) or maximum size reached.
+        if (seg.buffer() == null) {
+            seg.release();
+
+            if (!fileWriter.isCancelled())
+                fileWriter.onMaxFileSizeReached();
+
+            return null;
+        }
+
+        ByteBuffer buf = seg.buffer();
+
+        buf.put((byte)type.ordinal());
+
+        return seg;
+    }
+
+    /** @return Performance statistics file. */
+    public static File statisticsFile(GridKernalContext ctx) throws IgniteCheckedException {
+        String igniteWorkDir = U.workDirectory(ctx.config().getWorkDirectory(), ctx.config().getIgniteHome());
+
+        File fileDir = U.resolveWorkDirectory(igniteWorkDir, PERFORMANCE_STAT_DIR, false);
+
+        return new File(fileDir, "node-" + ctx.localNodeId() + ".prf");
+    }
+
+    /** Writes {@link UUID} to buffer. */
+    public static void writeUuid(ByteBuffer buf, UUID uuid) {
+        buf.putLong(uuid.getMostSignificantBits());
+        buf.putLong(uuid.getLeastSignificantBits());
+    }
+
+    /** Reads {@link UUID} from buffer. */
+    public static UUID readUuid(ByteBuffer buf) {
+        return new UUID(buf.getLong(), buf.getLong());
+    }
+
+    /** Writes {@link IgniteUuid} to buffer. */
+    public static void writeIgniteUuid(ByteBuffer buf, IgniteUuid uuid) {
+        buf.putLong(uuid.globalId().getMostSignificantBits());
+        buf.putLong(uuid.globalId().getLeastSignificantBits());
+        buf.putLong(uuid.localId());
+    }
+
+    /** Reads {@link IgniteUuid} from buffer. */
+    public static IgniteUuid readIgniteUuid(ByteBuffer buf) {
+        UUID globalId = new UUID(buf.getLong(), buf.getLong());
+
+        return new IgniteUuid(globalId, buf.getLong());
+    }
+
+    /** Worker to write to performance statistics file. */
+    private class FileWriter extends GridWorker {
+        /** Maximum cached string count. */
+        private static final short MAX_CACHED_STRING_COUNT = Short.MAX_VALUE;
```

Comment:
This constant not used now. Please, remove it.

Reply:
Done.